deploy/vm/template.json

```diff
@@ -951,6 +1041,14 @@
         "resourceGroup" : {
             "type": "string",
             "value": "[resourceGroup().name]"
-        }       
+        },
+        "appinsights-name": {
```

Comment:
You need to also update the setup.sh file (as well as calling it) with the instrumentation key and id.   The setup.sh is called to setup the vm, as part of the arm template.   It is in the same folder.